### PR TITLE
Add FTS Agent

### PR DESCRIPTION
**Pull request by kmharrington**
Add an Agent for controlling the FTS.

## Description
The agent can be used to control the two UChicago FTSs. One will be used with the LATRt and the other will be used at K2SO. They will probably also make it to the site at some point.

## How Has This Been Tested?
This has been tested with one of the FTSs for a complete measurement (moving the FTS through a range of positions while taking data). 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:191]: You should handle the case that `params` didn't get passed (whether intentionally or not).

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:197]: Is there a sensible default position? If `params` gets passed but without the `'position'` key, this'll pass `None` to `self.stage.move_to()`, which I don't think handles that case.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:199]: Maybe return the result of `self.stage.move_to()` here? It looks like if I pass an out of bounds position, `self.stag.move_to()` will return `False, "Move out of bounds!"`, but the task here then returns `True, "Move Complete"`. And I think the result would be the stage wouldn't have moved, but the task would report moving.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:182]: What's the response read after sending the `HOME X` command? Should it checked here to make sure it was successful?

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:164]: I think this would run into the same scenario that you encountered in #164. A fix was implemented in #168 that you could adapt here to handle this case.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:243]: If `get_position()` isn't successful this is going to put a `None` into the data array, which will cause a `TypeError` when published. We should prevent that from happening. Perhaps check the success state before publishing?

File: docs/agents/fts_agent.rst
Comment (BrianJKoopman) [inline docs/agents/fts_agent.rst:10]: I know there are other FTS in the collaboration. Is there a good way to distinguish them? Is it by motion controller? Inevitably we'll probably run into naming choices with the Agents corresponding to this and the other FTS. Is there a better, more specific name we could choose?

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:73]: What's this bare `except` catching?

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:180]: The f-string gets broken by that line break. Move the `f` down to the second string.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:195]: Same f-string problem as above.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:223]: Same f-string problem as above.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [inline agents/fts_stage/fts_agent.py:232]: Same f-string problem as above.

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
Thanks for the PR, Katie! I think there are a few edge cases we need to preemptively handle before this is ready to merge. I'm also curious to hear your thoughts on naming when considering there will be other FTS Agents for the other FTS that I think is at UCSD.

File: agents/fts_stage/fts_agent.py
Comment (kmharrington) [reply agents/fts_stage/fts_agent.py:73]: Honestly no idea, that part of the code came from ACT's "agent." Looking at it I'm guessing it's protecting from getting a non-sense response from the controller.

File: agents/fts_stage/fts_agent.py
Comment (BrianJKoopman) [reply agents/fts_stage/fts_agent.py:164]: Yup, that's it. The following return is still relevant too, and completes the task/process (by saying that it failed.)

File: docs/agents/fts_agent.rst
Comment (kmharrington) [reply docs/agents/fts_agent.rst:10]: That is a good point. We're sending our second FTS to UCSD and it should have the exact same controller as this one. So it should work the same but if there's small differences I'll help debug to get them both working on the same agent. As far as I know these two are the only two that could end up at the site? 

The other FTS I know about is the one being used at K2SO. I don't know if @jseibert575  has plans to make an OCS agent for that one?

File: agents/fts_stage/fts_agent.py
Comment (kmharrington) [reply agents/fts_stage/fts_agent.py:182]: It returns nothing useful. The only thing is the read call blocks until the home is done. And since the controller ignores everything while it's running I have that read call there mostly just to block.

**Comment by kmharrington** [comment]
Ok! I think I've got everything updated off your review!

**Review by BrianJKoopman — APPROVED** [review]
Thanks Katie, looks good! One quick question on the naming, is Aerotech the brand of the stage? Before I merge I'm going to rebase and handle some conflicts that arose, but am also thinking I'll rename the directory this is in from `fts_stage` to `fts_aerotech` or similar. Any better suggestions? (Just wondering if `fts_stage` is still general enough to potentially conflict with another FTS Agent.)

**Comment by kmharrington** [comment]
Yeah Aerotech is the company name for the controller. I left the folder as-is kind of thinking we could have multiple agents in there for different FTSs. But that's totally up to you.

**Comment by BrianJKoopman** [comment]
> Yeah Aerotech is the company name for the controller. I left the folder as-is kind of thinking we could have multiple agents in there for different FTSs. But that's totally up to you.

I'd like to keep them separate, especially if they end up being added to the Docker builds. I moved to `fts_aerotech_stage/` and updated the needed paths in files. Also caught up to the latest develop and handled those few merge conflicts. Will merge when checks finish.

**Pull request merged**
